safe/impact_function/multi_exposure_wrapper.py

```diff
@@ -446,3 +470,55 @@ def run(self):
             self._analysis_summary, analysis_width, analysis_color)
 
         return ANALYSIS_SUCCESS, None
+
+    def generate_report(self, components, output_folder=None, iface=None):
+        """Generate Impact Report independently by the Impact Function.
+
+        :param components: Report components to be generated.
+        :type components: list
+
+        :param output_folder: The output folder.
+        :type output_folder: str
+
+        :returns: Tuple of error code and message
+        :type: tuple
+
+        .. versionadded:: 4.3
+        """
+        # iface set up, in case IF run from test
+        if not iface:
+            iface = IFACE
+
+        for component in components:
+
+            report_metadata = ReportMetadata(
+                metadata_dict=update_template_component(component))
+
+            self.impact_report = ImpactReport(
```

Comment:
PyCharm is warning you about this variable not in __init__. Add it there
Question, should it be private?

Reply:
yes it should be private